### ISSUE TITLE
Remove redundant StudentRiskLevel records and add constraint/validation

### ISSUE THREAD

**Issue by kevinrobinson** [studentinsights/studentinsights#1074]
`StudentRiskLevel` is the largest table in the database, and there are ~100x times as many records there as there are students.

Here's what I see with a sample of 100 student records:
```
# get a sample
ss = Student.all.sample(50)

# how many student risk levels are there for each student in the sample?
ss.map {|s| StudentRiskLevel.where(student_id: s.id).size }.sort
=> [1, 1, 2, 2, 2, 3, 3, 3, 11, 13, 14, 15, 16, 17, 17, 17, 27, 28, 30, 38, 118, 122, 124, 131, 133, 137, 138, 138, 139, 140, 140, 140, 141, 142, 143, 144, 145, 146, 155, 157, 165, 168, 170, 172, 173, 181, 182, 184, 188, 194]
```

So this means:
- there is an active bug in the import process, or there was in the past and we didn't clean up the extra records
- there's no constraints or validations ensuring that each student only has one risk level

**Comment by kevinrobinson** [comment]
It seems like the behavior of `#create_student_risk_level!` is a bit complicated, and not what I expected.  Rails injects this method when the association is created.  Calling it when there already is a risk model raises, but first it deletes the existing `StudentRiskLevel`:

```
irb(main):001:0> s = Student.first.reload
=> #<Student id: 1, grade: "3", ... risk_level: 3, gender: "M">
irb(main):002:0> s.student_risk_level
=> #<StudentRiskLevel id: 91, student_id: 1, level: 3, created_at: "2017-09-24 16:52:25", updated_at: "2017-09-24 16:52:25", mcas_math_risk_level: 3, star_math_risk_level: 1, mcas_ela_risk_level: 0, star_reading_risk_level: 1, limited_english_proficiency_risk_level: nil>
irb(main):003:0> s.create_student_risk_level!
   (0.4ms)  BEGIN
  StudentRiskLevel Exists (0.6ms)  SELECT  1 AS one FROM "student_risk_levels" WHERE "student_risk_levels"."student_id" = $1 LIMIT $2  [["student_id", 1], ["LIMIT", 1]]
   (0.3ms)  ROLLBACK
   (0.2ms)  BEGIN
  SQL (0.5ms)  DELETE FROM "student_risk_levels" WHERE "student_risk_levels"."id" = $1  [["id", 91]]
   (2.0ms)  COMMIT
ActiveRecord::RecordInvalid: Validation failed: Student has already been taken
	from (irb):3

```
Okay, so it raises an error for the validation.  But what's that DELETE command about?  And if it deleted the existing record, why did the validation fail?

```
irb(main):004:0> s.student_risk_level
=> #<StudentRiskLevel id: nil, student_id: 1, level: nil, created_at: nil, updated_at: nil, mcas_math_risk_level: nil, star_math_risk_level: nil, mcas_ela_risk_level: nil, star_reading_risk_level: nil, limited_english_proficiency_risk_level: nil>
```

Huh so it has a `StudentRiskLevel` that hasn't been persisted (probably because the validation failed).  And in the meantime, the existing record was deleted, so this student probably has no risk level anymore, let's check:

```
irb(main):005:0> s = Student.first.reload
...
=> #<Student id: 1, grade: "3", ... risk_level: 3, gender: "M">
irb(main):006:0> s.student_risk_level
  StudentRiskLevel Load (0.5ms)  SELECT  "student_risk_levels".* FROM "student_risk_levels" WHERE "student_risk_levels"."student_id" = $1 LIMIT $2  [["student_id", 1], ["LIMIT", 1]]
=> nil
```

Yep, no risk level and the cached value on the student model is out of sync now, since that wasn't updated.  What happens if we create one now?

```
irb(main):007:0> s.create_student_risk_level!
...
=> #<StudentRiskLevel id: 92, student_id: 1, level: 3, created_at: "2017-09-24 16:54:49", updated_at: "2017-09-24 16:54:49", mcas_math_risk_level: 3, star_math_risk_level: 1, mcas_ela_risk_level: 0, star_reading_risk_level: 1, limited_english_proficiency_risk_level: nil>
...
```

So basically, calling `student.create_student_risk_level!` is not idempotent, and cycles between doing what you'd expect and deleting the existing record non-transactionally and then raising because of a conflict that isn't really a conflict.  I think next step is to look at the Rails source to understand more.

Alternately, within `Student#update_risk_level!`, let's just say explicitly what we're doing instead.  That's in part of https://github.com/studentinsights/studentinsights/pull/1089.

**Comment by kevinrobinson** [comment]
Closed out by https://github.com/studentinsights/studentinsights/pull/1089
